CIPs/CIP-69/CIP-69.md

```diff
@@ -0,0 +1,108 @@
+---
+cip: 69
+title: Batched Anchor Data Structure
+author: Joel Thorstensson (@oed)
+discussions-to: https://github.com/ceramicnetwork/CIP/issues/69
+status: Last Call
+category: Standards
+type: Core
+created: 2020-11-02
+---
+
+## Simple Summary
+This CIP describes the data structure of a blockchain anchor that batches multiple document updates together. 
+
+
+## Abstract
+An anchor service accepts document update anchor request from clients and batches them together in a single merkle tree, which is then anchored into a specific blockchain. The anchor service also sorts all of the requests and creates a bloom filter which describes which documents where updated in this batch, which is also included in the blockchain transaction. This allows external observers to more effectively determine if a document which they are interested in was updated.
+
+
+## Motivation
+There are two main motivations for this CIP. Firstly having an agreed upon spec for the structure of the batched anchor merkle tree makes it easier for all consumers of the ceramic protocol to know what to expect when parsing an *anchor record*. Secondly the sorting of updates and the inclusion of a bloom filter allows indexing services to effectively determine if an update was made which is relevant, and find this update in the tree effectively.
+
+
+## Specification
+
+First lets define the basic IPLD data structures used.
+```ipldsch
+type MerkleNode [Link]
+
+type TreeMetadata struct {
+  numEntries Int
+  bloomType String
+  bloomFilter {String:Any}
```

Comment:
Can't seem to make a mult-line suggestion in github but how about instead of separate top-level `bloomType` and `bloomFilter` fields we have a single top-level `bloomFilter` field whose value is an object with a `type` and a `data` field?  So then we can add more tree metadata in the future and it's not mixed-in with bloom filter specific data

Reply:
Sure.